TestDerivs/src/test.cxx

```diff
@@ -110,64 +182,170 @@ extern "C" void TestDerivs_CalcDerivs(CCTK_ARGUMENTS) {
-            dyydchi - gf_ddchi(1, 1)(p.I) > tiny ||
-            dyzdchi - gf_ddchi(1, 2)(p.I) > tiny ||
-            dzzdchi - gf_ddchi(2, 2)(p.I) > tiny) {
+        if (dchi(0) - gf_dchi(0)(p.I) > tiny ||
+            dchi(1) - gf_dchi(1)(p.I) > tiny ||
+            dchi(2) - gf_dchi(2)(p.I) > tiny ||
+            ddchi(0, 0) - gf_ddchi(0, 0)(p.I) > tiny ||
+            ddchi(0, 1) - gf_ddchi(0, 1)(p.I) > tiny ||
+            ddchi(0, 2) - gf_ddchi(0, 2)(p.I) > tiny ||
+            ddchi(1, 1) - gf_ddchi(1, 1)(p.I) > tiny ||
+            ddchi(1, 2) - gf_ddchi(1, 2)(p.I) > tiny ||
+            ddchi(2, 2) - gf_ddchi(2, 2)(p.I) > tiny) {
 #ifndef SYCL_LANGUAGE_VERSION
-          printf("ddxx = %f\n", dxxdchi - gf_ddchi(0, 0)(p.I));
-          printf("ddxy = %f\n", dxydchi - gf_ddchi(0, 1)(p.I));
-          printf("ddxz = %f\n", dxzdchi - gf_ddchi(0, 2)(p.I));
-          printf("ddyy = %f\n", dyydchi - gf_ddchi(1, 1)(p.I));
-          printf("ddyz = %f\n", dyzdchi - gf_ddchi(1, 2)(p.I));
-          printf("ddzz = %f\n", dzzdchi - gf_ddchi(2, 2)(p.I));
+          printf("dx = %f\n", dchi(0) - gf_dchi(0)(p.I));
+          printf("dy = %f\n", dchi(1) - gf_dchi(1)(p.I));
+          printf("dz = %f\n", dchi(2) - gf_dchi(2)(p.I));
+          printf("ddxx = %f\n", ddchi(0, 0) - gf_ddchi(0, 0)(p.I));
+          printf("ddxy = %f\n", ddchi(0, 1) - gf_ddchi(0, 1)(p.I));
+          printf("ddxz = %f\n", ddchi(0, 2) - gf_ddchi(0, 2)(p.I));
+          printf("ddyy = %f\n", ddchi(1, 1) - gf_ddchi(1, 1)(p.I));
+          printf("ddyz = %f\n", ddchi(1, 2) - gf_ddchi(1, 2)(p.I));
+          printf("ddzz = %f\n", ddchi(2, 2) - gf_ddchi(2, 2)(p.I));
 #endif
           assert(0);
         }
       });
 #endif
 }
 
+extern "C" void TestDerivs_CalcError(CCTK_ARGUMENTS) {
+  DECLARE_CCTK_ARGUMENTSX_TestDerivs_CalcError;
+  DECLARE_CCTK_PARAMETERS;
+
+  const Arith::vect<CCTK_REAL, dim> dx(std::array<CCTK_REAL, dim>{
+      CCTK_DELTA_SPACE(0),
+      CCTK_DELTA_SPACE(1),
+      CCTK_DELTA_SPACE(2),
+  });
+
+  grid.loop_int_device<0, 0, 0>(
+      grid.nghostzones, [=] ARITH_DEVICE(const PointDesc &p) ARITH_INLINE {
+        Arith::vec<CCTK_REAL, dim> dchi;
+        Arith::smat<CCTK_REAL, dim> ddchi;
+        poly_derivs(kxx, kxy, kyz, p.x, p.y, p.z, dchi, ddchi);
+
+        dxchi_error(p.I) = dxchi(p.I) - dchi(0);
+        dychi_error(p.I) = dychi(p.I) - dchi(1);
+        dzchi_error(p.I) = dzchi(p.I) - dchi(2);
+        dxxchi_error(p.I) = dxxchi(p.I) - ddchi(0, 0);
+        dxychi_error(p.I) = dxychi(p.I) - ddchi(0, 1);
+        dxzchi_error(p.I) = dxzchi(p.I) - ddchi(0, 2);
+        dyychi_error(p.I) = dyychi(p.I) - ddchi(1, 1);
+        dyzchi_error(p.I) = dyzchi(p.I) - ddchi(1, 2);
+        dzzchi_error(p.I) = dzzchi(p.I) - ddchi(2, 2);
+
+        const int diss_order = deriv_order + 2;
+        const CCTK_REAL diss =
+            Arith::pown(-1, diss_order / 2.0 - 1.0) /
+            Arith::pown(2, diss_order) *
```

Comment:
since `pown` takes an `int` as its second arguemtn, doing floating point math here seem wrong (since it relies on implicit casts from `double` to `int`) and confusing. Rather use `diss_order / 2 - 1`.

Reply:
the change introduces a clang-format-19 change:  
 

```
diff --git a/TestDerivs/src/test.cxx b/TestDerivs/src/test.cxx
index 04ce209b..d919800b 100644
--- a/TestDerivs/src/test.cxx
+++ b/TestDerivs/src/test.cxx
@@ -336,8 +336,7 @@ extern "C" void TestDerivs_CalcError(CCTK_ARGUMENTS) {

         const int diss_order = deriv_order + 2;
         const CCTK_REAL diss =
-            Arith::pown(-1, diss_order / 2 - 1) /
-            Arith::pown(2, diss_order) *
+            Arith::pown(-1, diss_order / 2 - 1) / Arith::pown(2, diss_order) *
             poly_diss(kxx, kxy, kyz, p.x, p.y, p.z, diss_order, dx);
         chi_diss_error(p.I) = diss - chi_diss(p.I);
```